lib/src/dartdoc.dart

```diff
@@ -36,14 +37,45 @@ class DartdocFileWriter implements FileWriter {
   @override
   final Set<String> writtenFiles = {};
 
-  DartdocFileWriter(this._outputDir, this.resourceProvider);
+  final int? _maxFileCount;
+  final int? _maxTotalSize;
```

Comment:
If `0` means "no limit", then we don't need nullability. No reason to have two values that both mean "no limit".
```suggestion
  final int _maxFileCount;
  final int _maxTotalSize;
```